Mathlib/Data/Nat/BinaryRec.lean

```diff
@@ -0,0 +1,157 @@
+/-
+Copyright (c) 2022 Praneeth Kolichala. All rights reserved.
```

Comment:
I think this should read "2017 Microsoft Corporation", `binary_rec` came from https://github.com/leanprover-community/lean/blob/master/library/init/data/nat/bitwise.lean authored by @digama0.